File: backend/src/main/java/com/localai/chat/settings/UserSettings.java
package com.localai.chat.settings;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "settings")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserSettings {

    @Id
    private String id;

    @Indexed(unique = true)
    private String userId;

    @Builder.Default private String defaultModel = "llama3.1:8b";
    @Builder.Default private double temperature = 0.7;
    @Builder.Default private double topP = 0.9;
    @Builder.Default private int topK = 40;
    @Builder.Default private int maxTokens = 2048;

    @Builder.Default private boolean streamingEnabled = true;
    @Builder.Default private boolean markdownEnabled = true;
    @Builder.Default private boolean animationsEnabled = true;

    @Builder.Default private String theme = "system"; // light | dark | system
    @Builder.Default private String language = "en";
    @Builder.Default private String fontSize = "medium"; // small | medium | large
}

File: lib/features/settings/domain/user_settings.dart
class UserSettings {
  final String defaultModel;
  final double temperature;
  final double topP;
  final int topK;
  final int maxTokens;
  final bool streamingEnabled;
  final bool markdownEnabled;
  final bool animationsEnabled;
  final String theme;
  final String language;
  final String fontSize;

  const UserSettings({
    this.defaultModel = 'llama3.1:8b',
    this.temperature = 0.7,
    this.topP = 0.9,
    this.topK = 40,
    this.maxTokens = 2048,
    this.streamingEnabled = true,
    this.markdownEnabled = true,
    this.animationsEnabled = true,
    this.theme = 'system',
    this.language = 'en',
    this.fontSize = 'medium',
  });

  UserSettings copyWith({
    String? defaultModel,
    double? temperature,
    double? topP,
    int? topK,
    int? maxTokens,
    bool? streamingEnabled,
    bool? markdownEnabled,
    bool? animationsEnabled,
    String? theme,
    String? language,
    String? fontSize,
  }) {
    return UserSettings(
      defaultModel: defaultModel ?? this.defaultModel,
      temperature: temperature ?? this.temperature,
      topP: topP ?? this.topP,
      topK: topK ?? this.topK,
      maxTokens: maxTokens ?? this.maxTokens,
      streamingEnabled: streamingEnabled ?? this.streamingEnabled,
      markdownEnabled: markdownEnabled ?? this.markdownEnabled,
      animationsEnabled: animationsEnabled ?? this.animationsEnabled,
      theme: theme ?? this.theme,
      language: language ?? this.language,
      fontSize: fontSize ?? this.fontSize,
    );
  }

  factory UserSettings.fromJson(Map<String, dynamic> json) {
    return UserSettings(
      defaultModel: json['defaultModel'] as String? ?? 'llama3.1:8b',
      temperature: (json['temperature'] as num?)?.toDouble() ?? 0.7,
      topP: (json['topP'] as num?)?.toDouble() ?? 0.9,
      topK: (json['topK'] as num?)?.toInt() ?? 40,
      maxTokens: (json['maxTokens'] as num?)?.toInt() ?? 2048,
      streamingEnabled: json['streamingEnabled'] as bool? ?? true,
      markdownEnabled: json['markdownEnabled'] as bool? ?? true,
      animationsEnabled: json['animationsEnabled'] as bool? ?? true,
      theme: json['theme'] as String? ?? 'system',
      language: json['language'] as String? ?? 'en',
      fontSize: json['fontSize'] as String? ?? 'medium',
    );
  }

  Map<String, dynamic> toJson() => {
        'defaultModel': defaultModel,
        'temperature': temperature,
        'topP': topP,
        'topK': topK,
        'maxTokens': maxTokens,
        'streamingEnabled': streamingEnabled,
        'markdownEnabled': markdownEnabled,
        'animationsEnabled': animationsEnabled,
        'theme': theme,
        'language': language,
        'fontSize': fontSize,
      };
}
